instantsearch/src/main/java/com/algolia/instantsearch/ui/views/Hits.java

```diff
@@ -551,17 +555,37 @@ private Activity getActivity(View view) {
         }
 
         class ViewHolder extends RecyclerView.ViewHolder {
-            final Map<View, String> viewMap = new HashMap<>();
+            private final Map<View, String> viewMap = new HashMap<>();
+            // needed to differentiate initial key value from Pair{null,null} which is legitimate
+            private final String defaultValue = "ADefaultValueForHitsVariant";
 
             ViewHolder(@NonNull View itemView) {
                 super(itemView);
-                // Store every annotated view with its attribute name
-                final SparseArray<String> attributes = BindingHelper.getBindings();
-                for (int i = 0; i < attributes.size(); i++) {
-                    viewMap.put(itemView.findViewById(attributes.keyAt(i)), attributes.valueAt(i));
+
+                String indexVariant = defaultValue;
+
+                // Get the index and variant for this layout
+                final List<View> views = LayoutViews.findAny((ViewGroup) itemView);
+                for (View view : views) {
+                    if (view instanceof AlgoliaHitView) {
+                        continue;
+                    }
+                    String viewIndexVariant = BindingHelper.getVariantForView(view);
+                    final boolean isDifferentVariant = !(viewIndexVariant == null ? indexVariant == null : viewIndexVariant.equals(indexVariant));
```

Comment:
rename isSameVariant and remove the ! before. Add it to the next if. Feels easier to read

Reply:
Fair point.